dev-tools/packaging/templates/docker/docker-entrypoint.agent.tmpl

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -eo pipefail
+
+function setup(){
+  curl -X POST  ${KIBANA_HOST:-http://localhost:5601}/api/ingest_manager/setup -H 'kbn-xsrf: true' -u ${KIBANA_USERNAME:-elastic}:${KIBANA_PASSWORD:-changeme}
+  curl -X POST  ${KIBANA_HOST:-http://localhost:5601}/api/ingest_manager/fleet/setup \
+    -H 'Content-Type: application/json' \
+    -H 'kbn-xsrf: true' \
+    -d '{"admin_username":"elastic","admin_password":"changeme"}' \
+    -u ${KIBANA_USERNAME:-elastic}:${KIBANA_PASSWORD:-changeme}
+}
+
+function enroll(){
+    # enroll agent to kibana
+    local enrollResp
+    enrollResp=$(curl -X POST  ${KIBANA_HOST:-http://localhost:5601}/api/ingest_manager/fleet/enrollment-api-keys \
+      -H 'Content-Type: application/json' \
+      -H 'kbn-xsrf: true' \
+      -u ${KIBANA_USERNAME:-elastic}:${KIBANA_PASSWORD:-changeme} \
+      -d '{"name":"demotoken","config_id":"default"}')
+
+    local exitCode=$?
+    if [ $exitCode -ne 0 ]; then
+      exit $exitCode
+    fi
+
+    apikey=$(echo $enrollResp | jq -r '.item.api_key')
+    ./{{ .BeatName }} enroll ${KIBANA_HOST:-http://localhost:5601} $apikey -f
+}
+yum install -y epel-release
+yum install -y jq
+
+setup
+enroll
+
+exec {{ .BeatName }} run "$@"
```

Comment:
I think there is a small problem here,  for testing using a docker container this work great, the setup + enroll using credentials but in the normal scenario we would use an enrollment key and nothing else, no?

Reply:
yes this container serves testing first, to be usable in a normal use cases it needs some tweaking.  i was thinking about different options such as using flags to determine if we're testing or not or setting env variable like `SETUP_KIBANA` or `PERFORM_ENROLL` to perform any additional steps. without variables it would just do run.

Reply:
I have mixed feeling here, having something quickly working and adding a bit more work to make it behave with enroll? Do you think it is a lot of work to add it now?

Reply:
no i dont' but i'm not sure about which approach to take. flags vs env vars vs something else i dont see

Reply:
I think environment variables is fine for first version, I prefer we can the functionality right and we can change the flags until we move to GA.